### PR TITLE
Adds an optional configuration to print out the URL to the launch in Report Portal

### PR DESCRIPTION
**Pull request by dylan8902**
To make it nice and easy for developers and testers to see where things have gone right or wrong, an optional link to the launch report is printed out when available

A derivation of the work done here: https://github.com/reportportal/agent-ruby/pull/96

### CODE REVIEW

File: lib/parallel_report_portal/cucumber/report.rb [hunk 1]
```diff
@@ -50,39 +50,44 @@ def launch_started(start_time)
...
       def launch_finished(clock)
         @tree.postordered_each do |node|
-          ParallelReportPortal.req_feature_finished(node.content, clock) unless node.is_root?
+          response = ParallelReportPortal.req_feature_finished(node.content, clock) unless node.is_root?
+          link = get_report_link(response)
+        end
+        response = ParallelReportPortal.req_launch_finished(launch_id, clock)
+        link = get_report_link(response)
+        if ParallelReportPortal.configuration.log_launch_link
+          print "Launch in ReportPortal: #{link}"
```
Comment (BillyRuffian) [inline lib/parallel_report_portal/cucumber/report.rb:65]: This is a print and not a puts? Is the omission of a new line after this output deliberate?

Comment (dylan8902) [reply lib/parallel_report_portal/cucumber/report.rb:65]: image


Not deliberate, copy & pasted from the original PR - Seems happy as the print

File: lib/parallel_report_portal/cucumber/report.rb [hunk 2]
```diff
@@ -223,8 +228,14 @@ def status_to_level(status)
           LOG_LEVELS.fetch(status, LOG_LEVELS[:info])
         end
       end
-      
-    
+
+      def get_report_link(response)
```
Comment (BillyRuffian) [inline lib/parallel_report_portal/cucumber/report.rb:232]: I’d prefer this to be called `parse_report_link_from_response` or some such. `get_` prefixes on method names are rarely used but this is a useful thing to store.

**Comment by BillyRuffian** [comment]
I get what this is trying to do and I get its utility too. It does feel a little narrow though. Wouldn’t it be better to:

a) expose an accessor to the member which contains the link as part of the api
b) add a hook which takes a block, allowing it to execute an arbitrary code as part of launch closure

**Comment by BillyRuffian** [comment]
e.g. (something like)

```ruby
ReportPortal.after_launch do
  LOG.info { “Launch report for #{ReportPortal.configuration.launch} available at #{ReportPortal.report_url}” }
end
```

——

```ruby
def self.after_launch(&block)
  @launch_finished_block = block
end
```

——

```ruby
def launch_finished
  …
  @launch_finished_block.call if @launch_finshed_block
end
```

(From memory one and only one process in a parallel run will ever call `finish_launch`)

**Comment by JeCFe** [comment]
Thanks for the feedback! We've attempted to implemented those changes, would appericate anymore feedback :)

**Pull request merged**